### ISSUE TITLE
[BUG]: AppImage Does not Work in NixOS

### ISSUE THREAD

**Issue by Obikawa** [KomodoPlatform/komodo-wallet-desktop#1876]
I would like to request a more complete AppImage release.

I use a very exotic distribution of Linux. So far, I was able to use all the software I needed - both packaged and and distributed as AppImages. AtomicDEX is the first to resist my ( admittedly weak ) attempts to run it. The most useful piece of information that I was able to extract from the logs is that my version of QT ( 5.15.3 ) is incompatible with the program ( 5.15.2 ).

From this, I inducted that the AppImage is not self-sustainable enough. I am not capable enough to build from source, considering the additional complexity introduced by the Nix system. In my opinion, the best course of action is to add more of the dependencies to the AppImage, so it can run on more systems.

The first step to this goal would be inclusion of QT libraries in the AppImage file.

**Comment by smk762** [comment]
The trade off here is larger binaries and build times. As the usage of exotic operating systems is less common (and often those running them can build from source) we focus on the most commonly used operating systems.

You should be able to install QT 5.15.2 alongside your current install - I recommend https://aqtinstall.readthedocs.io/en/latest/ 
This might resolve launch issue :crossed_fingers:

**Comment by tonymorony** [comment]
Thank you for the issue opening and insights @Obikawa 

We'll definitely consider our bundling pipeline updating/enhanced to support wider scope of distros and will try our best to allocate resources for that in the next sprint

**Comment by Obikawa** [comment]
Is the following log useful in any way? Only today I realized that my issue could be actually unrelated to AppImages.

```
$ LC_ALL=C appimage-run atomicdex-desktop-0.5.6-beta-ubuntu.AppImage 
atomicdex-desktop-0.5.6-beta-ubuntu.AppImage installed in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58
[2022-09-17 13:18:55.997] [info] [main.prerequisites.hpp:337] Installing qt_message_handler
[2022-09-17 13:18:56.000] [warning] [main.prerequisites.hpp:92] QSslSocket: cannot resolve EVP_PKEY_base_id
[2022-09-17 13:18:56.000] [warning] [main.prerequisites.hpp:92] QSslSocket: cannot resolve SSL_get_peer_certificate
[2022-09-17 13:18:56.004] [info] [main.prerequisites.hpp:342] SSL: true OpenSSL 1.1.1g  21 Apr 2020 OpenSSL 3.0.5 5 Jul 2022
[13:18:56] [info] [global.utilities.cpp:256] [76435]: Logger successfully initialized
[13:18:56] [info] [main.prerequisites.hpp:120] [76435]: connecting signal SIGABRT to the signal handler
[13:18:56] [info] [main.prerequisites.hpp:222] [76435]: Init timezone db
[13:18:56] [info] [main.prerequisites.hpp:145] [76435]: wally successfully initialized
[13:18:56] [info] [main.prerequisites.hpp:154] [76435]: libsodium successfully initialized
[13:18:56] [info] [main.prerequisites.hpp:160] [76435]: cleaning previous mm2 instance
mm2: no process found
[13:18:56] [info] [main.prerequisites.hpp:245] [76435]: Checking for setup default themes - theme_path: /home/marek/.atomic_qt/themes original_theme_path: /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/share/assets/themes
[13:18:56] [info] [main.prerequisites.hpp:246] [76435]: copying default themes into directory: /home/marek/.atomic_qt/themes
[13:18:56] [info] [main.prerequisites.hpp:261] [76435]: Checking for setup default logo - logo_path: /home/marek/.atomic_qt/logo original_logo_path: /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/share/assets/logo
[13:18:56] [info] [main.prerequisites.hpp:264] [76435]: copying default logo into directory: /home/marek/.atomic_qt/logo
[13:18:56] [info] [main.prerequisites.hpp:275] [76435]: Checking for settings ini reconfiguration
[13:18:56] [info] [main.prerequisites.hpp:297] [76435]: reconfiguration for settings finished
[13:18:56] [info] [main.prerequisites.hpp:180] [76435]: initializing high dpi support
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: Could not find the Qt platform plugin "wayland" in ""
ATTENTION: default value of option mesa_glthread overridden by environment.
[13:18:56] [debug] [main.prerequisites.hpp:190] [76435]: physical dpi: 119.80333333333333
[13:18:56] [debug] [main.prerequisites.hpp:191] [76435]: logical dpi: 96
[13:18:56] [debug] [main.prerequisites.hpp:196] [76435]: height: 1598
[13:18:56] [debug] [main.prerequisites.hpp:190] [76435]: physical dpi: 125.28569659442725
[13:18:56] [debug] [main.prerequisites.hpp:191] [76435]: logical dpi: 96
[13:18:56] [debug] [main.prerequisites.hpp:193] [76435]: scale: 1
[13:18:56] [debug] [main.prerequisites.hpp:196] [76435]: height: 940
[13:18:56] [debug] [main.prerequisites.hpp:193] [76435]: scale: 1
[13:18:56] [info] [main.prerequisites.hpp:208] [76435]: dpi settings finished
[13:18:56] [warning] [internet.checker.service.cpp:190] [76435]: mm2 system not available skipping internet connectivity with it
[13:18:56] [info] [ip.checker.service.cpp:96] [76435]: ip_service_checker created
[13:18:56] [info] [raw.mm2.coins.cfg.hpp:244] [76435]: parse_raw_mm2_coins_file
[13:18:56] [info] [raw.mm2.coins.cfg.hpp:269] [76435]: successfully parsed: /home/marek/.atomic_qt/0.5.6/configs/coins.json
[13:18:56] [info] [raw.mm2.coins.cfg.hpp:270] [76435]: coins size mm2: 640
[13:18:56] [info] [mm2.service.cpp:262] [76435]: mm2_service created
[13:18:56] [info] [qt.portfolio.page.cpp:38] [76435]: portfolio_page created
[13:18:56] [info] [orderbook.scanner.service.cpp:29] [76435]: orderbook_scanner_service created
[13:18:56] [warning] [rpc.best.orders.cpp:44] [76435]: best orders result not available yet - probably seed node unsync
[13:18:56] [info] [band.provider.cpp:54] [76435]: start fetching oracle
[13:18:56] [info] [komodo.prices.provider.cpp:14] [76435]: komodo_prices_provider created
[13:18:56] [info] [komodo.prices.provider.cpp:35] [76435]: komodo price service tick loop
[13:18:56] [info] [komodo.prices.cpp:82] [76435]: url: https://prices.komodo.live:1313/api/v2/tickers?expire_at=600
[13:18:56] [info] [coingecko.wallet.charts.cpp:39] [76435]: coingecko_wallet_charts_service created
[13:18:56] [info] [qt.orders.widget.cpp:33] [76435]: qt_orders_widget created
[13:18:56] [info] [qt.trading.page.cpp:659] [76435]: clearing forms : set_current_trading_mode
[13:18:56] [info] [qt.trading.page.cpp:1462] [76435]: reset_fees
[13:18:56] [warning] [qt.trading.page.cpp:905] [76435]: last_trading_error is VolumeFieldNotFilled
[13:18:56] [debug] [qt.trading.page.cpp:949] [76435]: Set trading mode to: Simple
[13:18:56] [info] [app.cpp:484] [76435]: connecting signals
[13:18:56] [info] [qt.addressbook.page.cpp:52] [76435]: connecting addressbook signals
[13:18:56] [error] [cpprestsdk.utilities.cpp:51] [76464]: pplx task error: Error resolving address
[13:18:56] [info] [app.cpp:373] [76435]: application created
[13:18:56] [info] [main.prerequisites.hpp:315] [76435]: file name settings: /home/marek/.atomic_qt/0.5.6/configs/cfg.ini
[13:18:56] [info] [main.prerequisites.hpp:312] [76435]: Settings CurrentTheme already exist - skipping
[13:18:56] [info] [main.prerequisites.hpp:312] [76435]: Settings ThemePath already exist - skipping
[13:18:56] [info] [main.prerequisites.hpp:312] [76435]: Settings AutomaticUpdateOrderBot already exist - skipping
[13:18:56] [info] [main.prerequisites.hpp:312] [76435]: Settings WalletChartsCategory already exist - skipping
[13:18:56] [info] [main.prerequisites.hpp:312] [76435]: Settings CurrentLang already exist - skipping
[13:18:56] [info] [main.prerequisites.hpp:312] [76435]: Settings 2FA already exist - skipping
[13:18:56] [info] [main.prerequisites.hpp:312] [76435]: Settings MaximumNbCoinsEnabled already exist - skipping
[13:18:56] [info] [main.prerequisites.hpp:312] [76435]: Settings DefaultTradingMode already exist - skipping
[13:18:56] [info] [main.prerequisites.hpp:312] [76435]: Settings FontMode already exist - skipping
[13:18:56] [warning] [app.cpp:315] [76435]: AutomaticUpdateOrderBot is false, ignoring the service
[13:18:56] [info] [qt.portfolio.page.cpp:190] [76435]: new chart category: OneMonth
[13:18:56] [info] [main.prerequisites.hpp:312] [76435]: Settings AvailableLang already exist - skipping
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: Could not find the Qt platform plugin "wayland" in ""
ATTENTION: default value of option mesa_glthread overridden by environment.
[13:18:56] [info] [qt.settings.page.cpp:116] [76435]: Locale before parsing AtomicDEX settings: C
[13:18:56] [info] [qt.settings.page.cpp:118] [76435]: Locale after parsing AtomicDEX settings: en_US
[13:18:56] [info] [qt.settings.page.cpp:126] [76435]: Successfully loaded en translation in :/assets/languages/atomic_defi_en.qm
[13:18:56] [info] [main.prerequisites.hpp:398] [76435]: post set_qt_app
[13:18:56] [info] [main.prerequisites.hpp:416] [76435]: QML Enum created
[13:18:56] [info] [main.prerequisites.hpp:441] [76435]: QML context properties created
[13:18:56] [debug] [main.prerequisites.hpp:86] [76435]: Load Qaterial v1.4.5
[13:18:56] [info] [main.prerequisites.hpp:89] [76435]: Load font ":/Qaterial/Fonts/Lato/Lato-Regular.ttf"
[13:18:56] [info] [main.prerequisites.hpp:89] [76435]: Load font ":/Qaterial/Fonts/Roboto/Roboto-Medium.ttf"
[13:18:56] [info] [main.prerequisites.hpp:89] [76435]: Load font ":/Qaterial/Fonts/Roboto/Roboto-Regular.ttf"
[13:18:56] [info] [main.prerequisites.hpp:89] [76435]: Load font ":/Qaterial/Fonts/RobotoMono/RobotoMono-Regular.ttf"
[13:18:56] [info] [main.prerequisites.hpp:448] [76435]: Qaterial type created
[13:18:56] [info] [main.prerequisites.hpp:458] [76435]: QML singleton created
[13:18:56] [info] [main.prerequisites.hpp:471] [76435]: Load qml engine
[13:18:56] [warning] [main.prerequisites.hpp:92] [76531]: Cyclic dependency detected between "qrc:/Dex/Constants/DexTypo.qml" and "qrc:/Dex/Constants/DexTypo.qml"
[13:18:56] [error] [cpprestsdk.utilities.cpp:51] [76486]: pplx task error: Error in SSL handshake
[13:18:56] [info] [komodo.prices.provider.cpp:48] [76473]: komodo price registry size: 233
[13:18:56] [debug] [app.cpp:549] [76473]: on_fiat_rate_updated
[13:18:56] [info] [qt.portfolio.page.cpp:103] [76473]: Updating portfolio values with model: true
[13:18:56] [info] [qt.portfolio.model.cpp:105] [76473]: update_currency_values
[13:18:56] [info] [ip.checker.service.cpp:90] [76459]: my ip address is: [89.228.61.5]
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: QQmlApplicationEngine failed to load component
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: qrc:/Dex/App.qml:90:9: Type Dashboard unavailable
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: qrc:/Dex/Screens/Dashboard.qml:127:13: Type Portfolio unavailable
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: qrc:/Dex/main.qml:386:5: Type App unavailable
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: qrc:/Dex/Portfolio/Portfolio.qml:196:21: Type AmountChart unavailable
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: qrc:/Dex/Portfolio/AmountChart.qml:223:13: Type DefaultBusyIndicator unavailable
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: qrc:/Dex/Components/DefaultBusyIndicator.qml:7:1: Type DexBusyIndicator unavailable
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: file:///home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/qml/QtQuick/Controls.2/BusyIndicator.qml:40:1: plugin cannot be loaded for module "QtQuick.Templates": Cannot load library /nix/store/qa2bbkcb9l2mllidqjw9abqg0y26nz7b-qtquickcontrols2-5.15.5-bin/lib/qt-5.15.5/qml/QtQuick/Templates.2/libqtquicktemplates2plugin.so: (/nix/store/qa2bbkcb9l2mllidqjw9abqg0y26nz7b-qtquickcontrols2-5.15.5-bin/lib/qt-5.15.5/qml/QtQuick/Templates.2/libqtquicktemplates2plugin.so: undefined symbol: _Z32QQuickTemplates_initializeModulev, version Qt_5)
[13:18:56] [info] [main.prerequisites.hpp:486] [76435]: qml engine successfully loaded
[13:18:56] [info] [app.cpp:184] [76435]: Launch the application
[13:18:56] [warning] [main.prerequisites.hpp:92] [76435]: qrc:/Dex/Components/DexBusyIndicator.qml:8:1: Type BusyIndicator unavailable
[13:18:56] [debug] [app.cpp:635] [76435]: will quit app, prevent all threading event
[13:18:56] [info] [mm2.service.cpp:299] [76435]: mm2 signals successfully disconnected
[13:18:56] [info] [mm2.service.cpp:324] [76435]: mm2 didn't stop yet with rpc stop, stopping process manually
[13:18:56] [info] [mm2.service.cpp:348] [76435]: mm2 service fully destroyed
[13:18:56] [info] [mm2.service.cpp:295] [76435]: destroying mm2 service...
[13:18:56] [info] [main.prerequisites.hpp:216] [76435]: wally successfully cleaned
[13:18:56] [info] [coingecko.wallet.charts.cpp:44] [76435]: coingecko_wallet_charts_service destroyed
[13:18:56] [info] [qt.orders.widget.cpp:35] [76435]: qt_orders widget destroyed
```

**Comment by smk762** [comment]
Thanks. One line stands out
` Could not find the Qt platform plugin "wayland" in "" ATTENTION: default value of option mesa_glthread overridden by environment.`
Are you able to try switching to XORG and see if launches? We had a similar problem with someone on Ubuntu 22.04.

**Comment by Obikawa** [comment]
Log on X11; crashes before showing anything, like before:

```
$ LC_ALL=C appimage-run atomicdex-desktop-0.5.6-beta-ubuntu.AppImage 
atomicdex-desktop-0.5.6-beta-ubuntu.AppImage installed in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58
[2022-09-17 20:35:05.226] [info] [main.prerequisites.hpp:337] Installing qt_message_handler
[2022-09-17 20:35:05.228] [warning] [main.prerequisites.hpp:92] QSslSocket: cannot resolve EVP_PKEY_base_id
[2022-09-17 20:35:05.228] [warning] [main.prerequisites.hpp:92] QSslSocket: cannot resolve SSL_get_peer_certificate
[2022-09-17 20:35:05.230] [info] [main.prerequisites.hpp:342] SSL: true OpenSSL 1.1.1g  21 Apr 2020 OpenSSL 3.0.5 5 Jul 2022
[20:35:05] [info] [global.utilities.cpp:256] [9355]: Logger successfully initialized
[20:35:05] [info] [main.prerequisites.hpp:120] [9355]: connecting signal SIGABRT to the signal handler
Previous run crashed:
 0# 0x0000000000460403 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/AppRun
 1# 0x00007FEA7881CBF0
 2# 0x00007FEA78869BC7
 3# 0x00007FEA7881CB46
 4# 0x00007FEA788074B5
 5# 0x00007FEA7A2929BC
 6# 0x00007FEA7A2B1874
 7# 0x00007FEA78F863F9
 8# 0x00007FEA7881F0C5
 9# 0x00007FEA7881F24E
10# 0x00000000004604CF in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/AppRun
11# 0x00007FEA7881CBF0
12# 0x00007FEA78869BC7
13# 0x00007FEA7881CB46
14# 0x00007FEA788074B5
15# 0x00000000004602E1 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/AppRun
16# 0x00007FEA7A2A9FE0
17# 0x00007FEA7A2929A4
18# 0x00007FEA7A4C852D
19# 0x00007FEA78FA2773
20# 0x00007FEA78FA37A4
21# 0x00007FEA78F8A379
22# 0x00007FEA78F8796A
23# 0x00007FEA78F88CC1
24# 0x00007FEA7A4CCDA3
25# 0x00007FEA7A49D7E9
26# 0x00007FEA7A4A0882
27# 0x00007FEA7A4F9B33
28# 0x00007FEA7C31CDFB
29# 0x00007FEA7C31D0A8
30# 0x00007FEA7C31D15F
31# 0x00007FEA7A4F91CC
32# 0x00007FEA7A49C21A
33# 0x00007FEA7A2B2844
34# 0x00007FEA78F864F7
35# 0x00007FEA7A2B3B35
36# 0x00007FEA78867E86
37# 0x00007FEA788EEC60

[20:35:05] [info] [main.prerequisites.hpp:222] [9355]: Init timezone db
[20:35:05] [info] [main.prerequisites.hpp:145] [9355]: wally successfully initialized
[20:35:05] [info] [main.prerequisites.hpp:154] [9355]: libsodium successfully initialized
[20:35:05] [info] [main.prerequisites.hpp:160] [9355]: cleaning previous mm2 instance
mm2: no process found
[20:35:05] [info] [main.prerequisites.hpp:245] [9355]: Checking for setup default themes - theme_path: /home/marek/.atomic_qt/themes original_theme_path: /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/share/assets/themes
[20:35:05] [info] [main.prerequisites.hpp:246] [9355]: copying default themes into directory: /home/marek/.atomic_qt/themes
[20:35:05] [info] [main.prerequisites.hpp:261] [9355]: Checking for setup default logo - logo_path: /home/marek/.atomic_qt/logo original_logo_path: /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/share/assets/logo
[20:35:05] [info] [main.prerequisites.hpp:264] [9355]: copying default logo into directory: /home/marek/.atomic_qt/logo
[20:35:05] [info] [main.prerequisites.hpp:275] [9355]: Checking for settings ini reconfiguration
[20:35:05] [info] [main.prerequisites.hpp:297] [9355]: reconfiguration for settings finished
[20:35:05] [info] [main.prerequisites.hpp:180] [9355]: initializing high dpi support
ATTENTION: default value of option mesa_glthread overridden by environment.
[20:35:05] [error] [main.prerequisites.hpp:98] [9507]: Cannot mix incompatible Qt library (5.15.5) with this library (5.15.2)
[20:35:05] [error] [main.prerequisites.hpp:106] [9507]: sigabort received, cleaning mm2
mm2: no process found
[20:35:05] [error] [main.prerequisites.hpp:112] [9507]: stacktrace:  0# 0x0000000000460403 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/AppRun
 1# 0x00007F5C7B81CBF0 in /usr/lib/libc.so.6
 2# 0x00007F5C7B869BC7 in /usr/lib/libc.so.6
 3# raise in /usr/lib/libc.so.6
 4# abort in /usr/lib/libc.so.6
 5# 0x00000000004602E1 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/AppRun
 6# 0x00007F5C7D2A9FE0 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
 7# QMessageLogger::fatal(char const*, ...) const in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
 8# 0x00007F5C7D4C852D in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
 9# 0x00007F5C7D09E773 in /nix/store/jfcdQThread::wait: Thread tried to wait on itself
m4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
10# QDBusAbstractInterface::QDBusAbstractInterface(QString const&, QString const&, char const*, QDBusConnection const&, QObject*) in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
11# QDBusConnectionInterface::QDBusConnectionInterface(QDBusConnection const&, QObject*) in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
12# 0x00007F5C7D08396A in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
13# 0x00007F5C7D084CC1 in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
14# QObject::event(QEvent*) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
15# QCoreApplication::notifyInternal2(QObject*, QEvent*) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
16# QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
17# 0x00007F5C7D4F9B33 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
18# g_main_context_dispatch in /usr/lib/libglib-2.0.so.0
19# 0x00007F5C7F31D0A8 in /usr/lib/libglib-2.0.so.0
20# g_main_context_iteration in /usr/lib/libglib-2.0.so.0
21# QEventDispatcherGlib::procesQMutex: destroying locked mutex
sEvents(QFlags<QEventLoop::ProcessEventsFlag>) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
QMutex: destroying locked mutex
22# QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
QThread: Destroyed while thread is still running
23# QThread::exec() in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
24# 0x00007F5C7D0824F7 in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
25# 0x00007F5C7D2B3B35 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
26# 0x00007F5C7B867E86 in /usr/lib/libc.so.6
27# 0x00007F5C7B8EEC60 in /usr/lib/libc.so.6

[20:35:05] [error] [main.prerequisites.hpp:106] [9507]: sigabort received, cleaning mm2
mm2: no process found
[20:35:05] [error] [main.prerequisites.hpp:112] [9507]: stacktrace:  0# 0x0000000000460403 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/AppRun
 1# 0x00007F5C7B81CBF0 in /usr/lib/libc.so.6
 2# 0x00007F5C7B869BC7 in /usr/lib/libc.so.6
 3# raise in /usr/lib/libc.so.6
 4# abort in /usr/lib/libc.so.6
 5# qErrnoWarning(char const*, ...) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
 6# QThread::~QThread() in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
 7# 0x00007F5C7D0823F9 in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
 8# 0x00007F5C7B81F0C5 in /usr/lib/libc.so.6
 9# 0x00007F5C7B81F24E in /usr/lib/libc.so.6
10# 0x00000000004604CF in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/AppRun
11# 0x00007F5C7B81CBF0 in /usr/lib/libc.so.6
12# 0x00007F5C7B869BC7 in /usr/lib/libc.so.6
13# raise in /usr/lib/libc.so.6
14# abort in /usr/lib/libc.so.6
15# 0x00000000004602E1 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/AppRun
16# 0x00007F5C7D2A9FE0 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
17# QMessageLogger::fatal(char const*, ...) const in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
18# 0x00007F5C7D4C852D in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
19# 0x00007F5C7D09E773 in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
20# QDBusAbstractInterface::QDBusAbstractInterface(QString const&, QString const&, char const*, QDBusConnection const&, QObject*) in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
21# QDBusConnectionInterface::QDBusConnectionInterface(QDBusConnection const&, QObject*) in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
22# 0x00007F5C7D08396A in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
23# 0x00007F5C7D084CC1 in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
24# QObject::event(QEvent*) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
25# QCoreApplication::notifyInternal2(QObject*, QEvent*) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
26# QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
27# 0x00007F5C7D4F9B33 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
28# g_main_context_dispatch in /usr/lib/libglib-2.0.so.0
29# 0x00007F5C7F31D0A8 in /usr/lib/libglib-2.0.so.0
30# g_main_context_iteration in /usr/lib/libglib-2.0.so.0
31# QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
32# QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
33# QThread::exec() in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
34# 0x00007F5C7D0824F7 in /nix/store/jfcdm4ffz3rxsqhk10bp1lrbjxv0iicn-qtbase-5.15.5-bin/lib/qt-5.15.5/plugins/platforms/../../../../../634m9rf1xclq7vdcpgqfqaiahgwdjkfy-qtbase-5.15.5/lib/libQt5DBus.so.5
35# 0x00007F5C7D2B3B35 in /home/marek/.cache/appimage-run/1345708d87e6e68b969aedc66968f0ba2956719803f22cf81cc8fd914f633d58/usr/bin/../lib/libQt5Core.so.5
36# 0x00007F5C7B867E86 in /usr/lib/libc.so.6
37# 0x00007F5C7B8EEC60 in /usr/lib/libc.so.6
```

**Comment by smk762** [comment]
Thanks, looks like another hint perhaps - `[20:35:05] [error] [main.prerequisites.hpp:98] [9507]: Cannot mix incompatible Qt library (5.15.5) with this library (5.15.2)`
@SylEze do you think this might work if there was no WebEngineView?

**Comment by Obikawa** [comment]
Could I encourage resolution of this issue with a donation of BTC?

**Comment by Obikawa** [comment]
Thank you for the response. My personal hypothesis is that the problem could be resolved by including some extra QT libraries in the AppImage. Does it make sense to you as a programmer?

```
[20:35:05] [error] [main.prerequisites.hpp:98] [9507]: Cannot mix incompatible Qt library (5.15.5) with this library (5.15.2)
```

Or does the new version use QT 5.15.5 and is able to mix with my system...

Is adding a couple more dependencies to AppImage so ... expensive, that it is to be avoided?

I have also looked again at the tools available and usable to me. NixOS' `appimage-run` has a switch to work with the "portable" format. Here is its output.

```
$ appimage-run -dw .
+ getopts x:w:dh option
+ case "${option}" in
+ export APPDIR=.
+ APPDIR=.
+ wrap_opt=true
+ getopts x:w:dh option
+ shift 2
+ '[' -n true ']'
+ '[' -d . ']'
+ wrap
+ export APPIMAGE_SILENT_INSTALL=1
+ APPIMAGE_SILENT_INSTALL=1
+ '[' -n '' ']'
+ exec ./AppRun
./AppRun: error while loading shared libraries: libQt5Charts.so.5: cannot open shared object file: No such file or directory
```

I would love to participate in the RC testing if it would result in improving the portability of the program.

**Comment by smk762** [comment]
We'll be looking to expand OS support in 0.5.8 - there are also a bunch of updates on the horizon including using qt6 which may be a good time to implement this

**Comment by Obikawa** [comment]
I had one more thought on this topic - I could try taking the missing libraries from whatever distribution the Appimage is made on.

Which distribution is the build host of the Appimage?

**Comment by Obikawa** [comment]
I have since given up on NixOS and AppImages as technologies, so this issue is no longer relevant to me.

I am currently working on packaging the Komodo Wallet as a Guix package and had reported on it in #2187 , which is in fact the source of the problem in this report. I am thinking it would be prudent to clean up after myself, so I am closing this issue.

**Comment by Obikawa** [comment]
Sorry about noise, I managed to fail closing the issue.
